Updated Forward Pixel Geometry: flex cables, CF sheet, support arms, cooling tubes

schneiml opened the pull request:
This PR supersedes #16316 .
- Commit 7f6d1c3 (thanks @Michael-Krohn !) but the dubious changes to the FPIX Polycone removed.
- Commit b222023 by myself to put the cable disks into the disk volumes.
- Commit 84246c7 with further changes by @tmulholland (thanks!)
- Update: Cooling tubes to disks added.
- Update: support arms added.

Still missing:
- extrusions due to Service cylinder volumes leaving the service cylinder. I'll try to fix this by growing it, there should be space for it. **Done**, CF sheet updated to real size. No overlaps now as far as I see.

To be fixed here or in another PR:
- Material mixtures should have the right components, but all mixtures/densities need to be checked.

@ianna : comments very welcome.
@davidlange6 : This PR seems ready to be merged to me. Update: still true, changing numbers can be a separate PR.

Also @veszpv , @friccita .

Michael-Krohn committed:
Updated the Forward Pixel Geometry to include Flex Cables, a Carbon Fiber sheet, and extended cabling

schneiml committed:
Move the disk of cables into the FPIX disks, to avoid overlaps and allow better placement.

Commit:
Additional FPIX changes by Troy.

* Flex cable location and size:
z="[pixfwdDisks:DiskHalfWidth]+[pixfwd:Disk3Z]+[FlexCableLength]/2+[SpaceBetweenCableBlocksandDisk3]"
I've set  [FlexCableLength] to (642*mm-[pixfwdDisks:DiskHalfWidth]) and
[SpaceBetweenCableBlocksandDisk3] to 4mm
this appears to be giving a slight overlap at the IP end of the flex
cables (see attached overlaps file). Should I shave them down a bit?

* Carbon fiber disk location:
z = [pixfwd:Disk1Z] - [pixfwdDisks:DiskHalfWidth] - [Spacing]
Here I've currently set [Spacing] to 2mm as a place holder. I'll change
this once you report back after checking the traveller document. This
2mm placeholder appears to be definitely giving overlaps. I'll recheck
for overlaps once we have the final measure.

* Flex cable density
density  6.615->3.3075

* Felx -> Flex (files changed):
1 occurrence in pixfwdCylinder.xml
3 occurrences in pixfwdMaterials.xml
1 occurrence in pixfwdOuterDiskZminus.xml
1 occurrence in pixfwdOuterDiskZplus.xml
1 occurrence in pixfwdbladeInnerZminus.xml
1 occurrence in pixfwdbladeInnerZplus.xml
1 occurrence in pixfwdbladeOuterZminus.xml
1 occurrence in pixfwdbladeOuterZplus.xml

cmsbuild commented:
A new Pull Request was created by  @schneiml (Marcel Schneider) for CMSSW_8_1_X.

It involves the following packages:

Geometry/TrackerCommonData


@civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please review it and eventually sign? Thanks.
@makortel, @ghellwig, @venturia, @VinInn, @ghugo83 this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

ianna commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

schneiml committed:
Grow the Service Cylinder Polycone to avoid extrusions

It now covers almost everything excluding the disks, touching disk 3 on the back and also the space between disk1 and BPIX, to fit the CF sheet there.

schneiml committed:
Marginally cut a support ring to avoid overlaps.

This overlap was caused by the slight movement of the OD rings to fit the cable disks inside the disk volumes.

cmsbuild commented:
Pull request #16338 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

schneiml commented on Geometry/TrackerCommonData/data/PhaseI/pixfwdCylinder.xml:
```diff
@@ -56,11 +56,20 @@ root volume plus an anchor point for positioning.
  <Constant name="CylindersServiceZMin" value="[CylindersServiceZoff]"/>
  <Constant name="CylindersServiceZMax" value="[CylindersOuterLength]"/>
+
+ <Constant name="CarbonFiberSheetNearIPRmin" value="101.0*mm"/>
```
@Michael-Krohn , @krose13 : is this 101mm a *real* value or one caused by limited size of the service cylinder? If this is only due to constraints, we can make it larger now.

cmsbuild commented:
-1

Tested at: 84246c7bd2dc6fa289332b1e760cb9a681215fcb

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15939/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
10021.0 step1
<pre>runTheMatrix-results/10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step1_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>10024.0 step1
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step1_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>

* **Material budget**:

There was error running material budget tests.

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

schneiml committed:
Use actual size for CF sheet, after a comment from Michael Krohn

cmsbuild commented:
Pull request #16338 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

schneiml commented:
@ianna the test complaint is fixed in 372b61c.

schneiml commented:
I will be mostly offline until tomorrow about an hour from now.

This PR is ok as far as I know, additional changes to add cooling material can be an independent PR (probably does not even conflict with this one).

I don't know what the plans of @krose13 , @tmulholland , @Michael-Krohn et. al. are for this; I (and in general people at CERN, as far as I know) do not have sufficient information to implement this.

ianna commented:
+1

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: cd135815bf7dd835f980775d5258ebf6429df51d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15951/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15951/summary.html

friccita commented:
From the before/after material budget comparison, I think that there is something seriously wrong with the masses of the newly-added objects. I suggest that the masses of these objects be checked first; perhaps the densities of the materials need to be corrected.

friccita commented:
Sorry for the alarm, I was mistaken in my previous comment. The material budget plots are for the entire Tracker, not just Pixels; and those peaks at |eta| < 2 were already there to begin with, so things are not as dark as I thought. Still, it would be good to know why the comparison plots did not get produced by the automated material budget test, and -- as a sanity check -- what the masses of the newly-added objects were.

cmsbuild commented:
+1
Tested at: cd135815bf7dd835f980775d5258ebf6429df51d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15986/summary.html

cmsbuild commented:
Comparison job queued.

schneiml commented:
@friccita A second set of tests (just completed) produced the plots and no error message.

Comparing the 3 Flex cable materials, I still think the new materials are too massive, due to the 100% aluminium.
But the cable volumes do represent a lot of material that is present in reality, so this is plausible and will cause changes.

schneiml commented on Geometry/TrackerCommonData/data/PhaseI/pixfwd.xml:
```diff
@@ -20,6 +20,10 @@
 
+ <Constant name="FlexCableDiskRmin" value="117.35*mm"/>
+ <Constant name="FlexCableDiskRmax" value="167.9*mm"/>
+ <Constant name="FlexCableDiskThickness" value="0.3*mm"/> <!-- This is actually *half* thickness -->
```
@Michael-Krohn @krose13 see the code comment that I added here. This might have to be halved, since the material used has already has a fairly high density and not all of the cable disk is actually covered by cable.

We have the question here whether we should include the mass of cables *outside* this volume in this volume (to get the total mass correct) or not. But here I think we should not, since we might move material from less critical regions right into the active area.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15986/summary.html

venturia commented:
Is it possible to sum the weights of all the volumes which are expected to simulate the cables from the modules to the portcards and let Marco Verzocchi know about it to allow him to compare with the actual weight?

schneiml committed:
Correct the Cable Disk Thickness after a comment of @krose13.

cmsbuild commented:
Pull request #16338 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 8230d1ff245a35871b02474bc32807c21a654181
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15997/summary.html

cmsbuild commented:
Comparison job queued.

ianna commented:
@venturia - here is a configurable analyzer which can give you weights (TID is set in the release):
```
cmsRun $CMSSW_RELEASE_BASE/src/SimG4Core/PrintGeomInfo/test/python/run_MaterialBudgeInfo_cfg.py
```
by element:
```
Element         Index   Total Mass      Mass Fraction   
N               0       45.1518 g       0.0011668
O               1       5.40257 kg      0.139612
Ar              2       777.232 mg      2.0085e-05
H               3       1.414 kg        0.0365402
C               4       18.5638 kg      0.479722
Na              5       29.0802 g       0.000751483
Al              6       5.07059 kg      0.131033
Si              7       3.35231 kg      0.0866296
K               8       35.7419 g       0.000923633
Fe              10      48.2663 g       0.00124729
Cr              12      6.4261 mg       1.66062e-07
Ni              13      19.4609 g       0.000502904
Cu              16      3.1794 kg       0.0821612
Sn              20      23.6621 g       0.000611471
B               22      12.8459 g       0.000331961
Ba              23      89.6561 g       0.00231687
Ti              24      46.6655 g       0.00120592
F               25      1.21734 kg      0.0314582
Sb              28      2.95239 g       7.62948e-05
Br              30      81.3177 g       0.00210139
Zn              31      28.9112 g       0.000747116
Ag              32      29.6277 g       0.000765632
B               33      2.91953 g       7.54458e-05

                Total Weight without Air 38.6971 kg     Total Fraction 1
TIDF.element (END) 
```
or by volume:
```
Geom.   Volume          Copy    Solid           Material        Density         Mass            
Level   Name            Number  Name            Name            [g/cm3]         [g]             
3       TIDF            1       TID             Air             1.214 mg/cm3    50.0073 g       
4       TIDSupport      1       TIDSupport      T_CarbonFibreStr        1.69 g/cm3      4.13131 kg      
5       TIDSupportIn    1       TIDSupportIn    T_Nomex         32 mg/cm3       234.678 g       
4       TIDSideDisk     1       TIDSideDisk     T_CarbonFibreStr        1.69 g/cm3      0 mg    
5       TIDSideDiskIn   1       TIDSideDiskIn   T_Nomex         32 mg/cm3       105.638 g       
4       TIDWheelF       3       TIDWheel        Air             1.214 mg/cm3    1.82704 g       
5       TIDRing0F       1       TIDRing         Air             1.214 mg/cm3    16.7441 g       
6       TIDStructure0   1       TIDStructure    TID_Mech        251.39 mg/cm3   1.28884 kg      
7       TIDRing0CoolManifold    2       TIDRing0CoolManifold    TID_CoolManifold        2.2075 g/cm3    60.044 g        
7       TIDRing0CoolManifold    1       TIDRing0CoolManifold    TID_CoolManifold        2.2075 g/cm3    60.044 g        
7       TIDModuleFix    98      TIDModuleFix    TID_ModuleFix   1.65714 g/cm3   998.001 mg      
7       TIDModuleFix    97      TIDModuleFix    TID_ModuleFix   1.65714 g/cm3   998.001 mg      
7       TIDModuleFix    96      TIDModuleFix    TID_ModuleFix   1.65714 g/cm3   998.001 mg      
7       TIDModuleFix    95      TIDModuleFix    TID_ModuleFix   1.65714 g/cm3   998.001 mg 
<snip>
```

ianna commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/15997/summary.html

schneiml committed:
Add the Support Arms.

- Material is a pure guess atm.
- Z position is incorrect, to avoid overlaps.
- phi positions are roughly read from the 3d CAD, should be ok to +-1mm.

ianna commented:
@schneiml - is total weight ~20 kg per each PixelForward expected? I run on 2017new geometry in CMSSW IB without this PR:
```
diff -wb PixelForwardZminus.element PixelForwardZplus.element
2,16c2,16
< N		0	103.968 g 	0.00542248
< O		1	2.33442 kg	0.121752
< H		3	442.369 g 	0.0230718
< C		4	7.75203 kg	0.404308
< Al		6	3.22228 kg	0.168058
< Si		7	965.063 g 	0.050333
< Cu		16	3.73453 kg	0.194775
< Sn		20	288.914 g 	0.0150684
< Cl		21	106.563 g 	0.0055578
< Sb		22	104.221 g 	0.00543565
< Pb		23	50.0855 g 	0.00261221
< Ba		24	32.2995 g 	0.00168458
< Ti		25	11.2601 g 	0.000587272
< B		26	22.0428 g 	0.00114964
< In		27	3.5427 g 	0.00018477
---
> N		0	103.968 g 	0.00542322
> O		1	2.33442 kg	0.121768
> H		3	442.369 g 	0.023075
> C		4	7.7494 kg	0.404226
> Al		6	3.22228 kg	0.168081
> Si		7	965.063 g 	0.0503399
> Cu		16	3.73453 kg	0.194801
> Sn		20	288.914 g 	0.0150704
> Cl		21	106.563 g 	0.00555856
> Sb		22	104.221 g 	0.00543639
> Pb		23	50.0855 g 	0.00261257
> Ba		24	32.2995 g 	0.00168481
> Ti		25	11.2601 g 	0.000587353
> B		26	22.0428 g 	0.0011498
> In		27	3.5427 g 	0.000184795
18c18
< 		Total Weight without Air 19.1736 kg	Total Fraction 1
---
> 		Total Weight without Air 19.171 kg	Total Fraction 1
```

schneiml committed:
Add Cooling Tubes inside the disk volume.

- Position in X/Y is approximate, but the real tubes are expected to be randomly curved anyways (bent by hand?)
- Position in Z is chosen to avoid collisions
- Only straight parts in the X/Y plane are modeled, length sometimes limted due to collisions
- Material is a guess

schneiml commented:
@ianna I have no idea. @krose13 might know. 

Just FYI: I will soon (~minutes) have more changes, adding more volumes (Cooling, Support arms). Shall I push these here or make a new PR?

schneiml committed:
Improve material mixtures.

These changes are *not* checked against reality. These are only guesses that should be more correct than what was there before, but all *numbers* have to be validatedagainst measurements/real material data.

cmsbuild commented:
Pull request #16338 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

schneiml commented:
I just added a number of new volumes (6 Support arms per disk, 8 cooling tubes per disk) and made some materials more relistic (where I am aware that the current mixture was incomplete).

![Disk](https://vgy.me/HqpN2f.jpg)

There are a batch of TODOs that need to be fixed by people with sufficient information. The output of the MatBudget tools (as described by @ianna ) above can help to check the densities. Be careful with the cooling tubes, the modelled sections to not map directly to physical parts; the total weight has to be spread in some way. Can somebody at Fermilab (@krose13, @tmulholland, @Michael-Krohn) take care of this?

Also, don't be misguided by the TODOs, there are likely also bugs in other materials. I just marked what I know is incorrect.

schneiml commented:
How to get component weights for material validation:
```
cmsrel CMSSW_8_1_0_pre15
cd CMSSW_8_1_0_pre15/src
cmsenv
git cms-merge-topic 16338
git cms-addpkg SimG4Core/PrintGeomInfo
sed -i -e s/TIDF/PixelForwardZplus/g \
  SimG4Core/PrintGeomInfo/test/python/run_MaterialBudgeInfo_cfg.py
scram b disable-biglib
cmsenv
scram b -j4
cmsRun SimG4Core/PrintGeomInfo/test/python/run_MaterialBudgeInfo_cfg.py
# full info in PixelForwardZplus.element, PixelForwardZplus.weight
# short summary
cat PixelForwardZplus.weight | cut -f2,6,7 | sort | uniq -c | column -t
```

Slightly annotated output:
```
#copies   Volume Name                     Density           Weight per copy

204   CarbonFiberSkin                     3.43511   g/cm3   473.459   mg
132   CarbonFiberSkin                     3.43511   g/cm3   481.67    mg
12    CoolingTubeToInnerDisk              1.5       g/cm3   601.157   mg
12    CoolingTubeToOuterDisk              1.5       g/cm3   326.34    mg
6     FlexCableDisk                       1.311     g/cm3   17.8164   g
204   PixelForwardAdhesiveFilm            2.87      g/cm3   339.999   mg  # inner/outer
132   PixelForwardAdhesiveFilm            2.93833   g/cm3   339.999   mg  # density properly adjusted
168   PixelForwardBlade                   1.214     mg/cm3
168   PixelForwardBladeBase               1.214     mg/cm3
5376  PixelForwardBumpBond                200.75    mg/cm3  0.517535  mg
5376  PixelForwardBumpROChip              1.214     mg/cm3
1     PixelForwardCarbonFiberSheetNearIP  134.087   mg/cm3  1.68018   g
1     PixelForwardCylinderElectronics1    277.14    mg/cm3  148.2     g
1     PixelForwardCylinderElectronics2    293.69    mg/cm3  182.64    g
1     PixelForwardCylinderElectronics3    293.69    mg/cm3  182.64    g
1     PixelForwardCylinderElectronics4    277.14    mg/cm3  148.2     g
2     PixelForwardCylinderEndFlange       1.66566   g/cm3   854.401   g
1     PixelForwardCylinderOuterCyl        134.087   mg/cm3  1.9704    kg
8     PixelForwardCylinderPipe1           1.31175   g/cm3   138.747   g
8     PixelForwardCylinderPipe2           3.05434   g/cm3   283.79    g
4     PixelForwardCylindersCoilFiber      585.59    mg/cm3  766.469   g
8     PixelForwardCylindersPortCards1     1.2349    g/cm3   834.575   g
3     PixelForwardDiskZplus               1.214     mg/cm3
336   PixelForwardEndHolder1              1.32      g/cm3
336   PixelForwardEndHolder2              1.32      g/cm3
132   PixelForwardFlexCable               2.463     g/cm3   54.311    mg
204   PixelForwardFlexCable               2.463     g/cm3   57.237    mg
8     PixelForwardFlexCables              3.3075    g/cm3   120.737   g
336   PixelForwardGlue                    1.1       g/cm3   136.378   mg
336   PixelForwardHDI                     1.90587   g/cm3   874.274   mg
336   PixelForwardHoldersConnector        1.214     mg/cm3
6     PixelForwardInnerDiskCFInnerRing    1.1744    g/cm3   2.01985   g
6     PixelForwardInnerDiskCFOuterRing    1.77      g/cm3   9.71078   g
6     PixelForwardInnerDiskInnerRing      1.566     g/cm3   14.0438   g
6     PixelForwardInnerDiskOuterRing      1.944     g/cm3   53.6913   g
336   PixelForwardModule                  1.214     mg/cm3
6     PixelForwardOuterDiskCFInnerRing    1.00003   g/cm3   3.00732   g
6     PixelForwardOuterDiskCFOuterRing    2.2677    g/cm3   18.2308   g
6     PixelForwardOuterDiskInnerRing      1.745     g/cm3   26.3185   g
6     PixelForwardOuterDiskOuterRing      1.612     g/cm3   54.3281   g
5376  PixelForwardROChip                  2.15758   g/cm3   37.0817   mg
336   PixelForwardSMKConn                 1.58      g/cm3   289.959   mg
336   PixelForwardSensor                  2.33      g/cm3
1     PixelForwardServiceCylinder         1.214     mg/cm3
336   PixelForwardWafer                   2.33      g/cm3
1     PixelForwardZplus                   1.214     mg/cm3
18    SupportArm                          1.495     g/cm3   3.38949   g  # fairly close to 10g/3 arms -> good
102   TPGBladeBase                        2.47      g/cm3   3.8583    g  # inner/outer
66    TPGBladeBase                        2.47      g/cm3   3.92522   g  # not adjusted
```

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

rovere commented:
@veszpv Do I understand correctly that the last commit from @schneiml can be considered as final?
I.e. nothing else is missing/left out?

schneiml commented:
@rovere I would not bet all on it, since some materials seem pretty far off (1.5x or more), but we need more information from FNAL to fix them. Yet it is unclear if and when this will happen.

The structure is the best we can/want to do, to my knowledge.

cmsbuild commented:
+1
Tested at: 4c7a18c3ff79f95789f0c85f5424e30c3420d887
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/16021/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/16021/summary.html

schneiml committed:
Adjust materials according to information given by Keith Rose.

In addition  to the densities,
- all CF-Materials were unified to be the same as the FPix_Cylinder_CF
- the Support Arms got a steel component (which is currently pure Fe and not fully correct)
- The Cooling pipe  steel got 2% Mo
The cooling pipe densities were averaged, since they use the same material

schneiml committed:
Separate Inner/Outer disk cooling tube materials, and adjust densities

cmsbuild commented:
Pull request #16338 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

schneiml commented:
PR updated with new materials with @krose13 's input.

New table:
```
204   CarbonFiberSkin                     3.43511   g/cm3   473.459   mg
132   CarbonFiberSkin                     3.43511   g/cm3   481.67    mg
12    CoolingTubeToInnerDisk              2.45      g/cm3   981.89    mg
12    CoolingTubeToOuterDisk              2.618     g/cm3   569.571   mg
6     FlexCableDisk                       160       mg/cm3  2.17439   g
204   PixelForwardAdhesiveFilm            2.87      g/cm3   339.999   mg
132   PixelForwardAdhesiveFilm            2.93833   g/cm3   339.999   mg
168   PixelForwardBlade                   1.214     mg/cm3
168   PixelForwardBladeBase               1.214     mg/cm3
5376  PixelForwardBumpBond                200.75    mg/cm3  0.517535  mg
5376  PixelForwardBumpROChip              1.214     mg/cm3
1     PixelForwardCarbonFiberSheetNearIP  134.087   mg/cm3  1.68018   g
1     PixelForwardCylinderElectronics1    526.53    mg/cm3  281.561   g
1     PixelForwardCylinderElectronics2    558.011   mg/cm3  347.016   g
1     PixelForwardCylinderElectronics3    558.011   mg/cm3  347.016   g
1     PixelForwardCylinderElectronics4    526.53    mg/cm3  281.561   g
2     PixelForwardCylinderEndFlange       1.66566   g/cm3   854.401   g
1     PixelForwardCylinderOuterCyl        134.087   mg/cm3  1.9704    kg
8     PixelForwardCylinderPipe1           1.31175   g/cm3   138.747   g
8     PixelForwardCylinderPipe2           3.05434   g/cm3   283.79    g
4     PixelForwardCylindersCoilFiber      398.21    mg/cm3  521.211   g
8     PixelForwardCylindersPortCards1     793.8     mg/cm3  536.469   g
3     PixelForwardDiskZplus               1.214     mg/cm3
336   PixelForwardEndHolder1              1.32      g/cm3
336   PixelForwardEndHolder2              1.32      g/cm3
132   PixelForwardFlexCable               2.463     g/cm3   54.311    mg
204   PixelForwardFlexCable               2.463     g/cm3   57.237    mg
8     PixelForwardFlexCables              2.55      g/cm3   93.0854   g
336   PixelForwardGlue                    1.1       g/cm3   136.378   mg
336   PixelForwardHDI                     1.90587   g/cm3   874.274   mg
336   PixelForwardHoldersConnector        1.214     mg/cm3
6     PixelForwardInnerDiskCFInnerRing    1.1744    g/cm3   2.01985   g
6     PixelForwardInnerDiskCFOuterRing    1.77      g/cm3   9.71078   g
6     PixelForwardInnerDiskInnerRing      1.566     g/cm3   14.0438   g
6     PixelForwardInnerDiskOuterRing      1.944     g/cm3   53.6913   g
336   PixelForwardModule                  1.214     mg/cm3
6     PixelForwardOuterDiskCFInnerRing    1.00003   g/cm3   3.00732   g
6     PixelForwardOuterDiskCFOuterRing    2.2677    g/cm3   18.2308   g
6     PixelForwardOuterDiskInnerRing      1.745     g/cm3   26.3185   g
6     PixelForwardOuterDiskOuterRing      1.612     g/cm3   54.3281   g
5376  PixelForwardROChip                  2.15758   g/cm3   37.0817   mg
336   PixelForwardSMKConn                 1.58      g/cm3   289.959   mg
336   PixelForwardSensor                  2.33      g/cm3
1     PixelForwardServiceCylinder         1.214     mg/cm3
336   PixelForwardWafer                   2.33      g/cm3
1     PixelForwardZplus                   1.214     mg/cm3
18    SupportArm                          1.495     g/cm3   3.38949   g
102   TPGBladeBase                        2.47      g/cm3   3.8583    g
66    TPGBladeBase                        2.47      g/cm3   3.92522   g
```

Some findings (Inner/Outer Blade inconsistencies) still hold, but are minor enough to be acceptable.

Total weight is now  18.5985 kg; This is a bit worrying, since we do expect around 22kg.

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 47ab05b4fba8631667be06c8ac7e28431758332a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/16066/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16338/16066/summary.html

ianna commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

davidlange6 commented:
+1

rovere commented:
@davidlange6 I'll submit a PR on top of this either late today or tomorrow with the tuned reco-material. From a quick investigation we have been lucky and apparently, out of the box, the agreement reco-simulation is already better wrt to the past.
More on this later on.
